test/test.py

```diff
@@ -728,6 +727,17 @@ def assert_bundle():
 
         assert_bundle()
 
+    def test_name_substitution(self):
+        uuid = 'ffee7f29-5c38-461a-8771-a68e20ec4a2e'
+        version = '2019-02-02T065454.662896Z'
+        manifest, metadata_files = self._load_bundle(uuid, version, replica='aws', deployment='prod')
+        bundle = Bundle(uuid, version, manifest, metadata_files)
+        for file_name in bundle.manifest:
+            self.assertNotIn('!', file_name)
```

Comment:
Not particularly strong. How do I know `!` was present initially and that it was actually replaced with `/`? It could have been absent initially or replaced with the wrong character.

Reply:
I've made the test more explicit and thorough.